presto-hive/src/main/java/com/facebook/presto/hive/security/ranger/RangerBasedAccessControl.java

```diff
@@ -0,0 +1,529 @@
+import static com.facebook.presto.hive.security.ranger.RangerBasedAccessControlConfig.RANGER_REST_USER_GROUP_URL;
+import static com.facebook.presto.hive.security.ranger.RangerBasedAccessControlConfig.RANGER_REST_USER_ROLES_URL;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyAddColumn;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyCreateSchema;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyCreateTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyCreateView;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyCreateViewWithSelect;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDeleteTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDropColumn;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDropSchema;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDropTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDropView;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyInsertTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyRenameColumn;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyRenameTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denySelectColumns;
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Connector access control which uses existing Ranger policies for authorizations
+ */
+
+public class RangerBasedAccessControl
+        implements ConnectorAccessControl
+{
+    private static final Logger log = Logger.get(RangerBasedAccessControl.class);
+
+    private RangerAuthorizer rangerAuthorizer;
+    private Users users;
+    private Map<String, Set<String>> userRolesMapping = new HashMap<>();
+    private Map<String, Set<String>> userGroupsMapping = new HashMap<>();
+
+    public RangerBasedAccessControl(RangerBasedAccessControlConfig config)
+    {
+        requireNonNull(config, "config is null");
+        requireNonNull(config.getRangerHttpEndPoint(), "Ranger service http end point is null");
+        requireNonNull(config.getRangerHiveServiceName(), "Ranger hive service name is null");
+
+        ServicePolicies servicePolicies;
+        try {
+            OkHttpClient client = getAuthHttpClient(config);
+            HttpUrl hiveServicePolicyUrl = requireNonNull(HttpUrl.get(URI.create(config.getRangerHttpEndPoint())))
+                    .newBuilder()
+                    .encodedPath(RANGER_REST_POLICY_MGR_DOWNLOAD_URL + "/" + config.getRangerHiveServiceName()).build();
+
+            HttpUrl getUsersUrl = requireNonNull(HttpUrl.get(URI.create(config.getRangerHttpEndPoint())))
```

Comment:
We should move these to `HttpClient` from com.facebook.airlift and that provides a lot of the configuration capabilities out of the box and should simplify our code

Reply:
@mayankgarg1990 Is there any existing implementation with `HttpClient` from com.facebook.airlift that I can refer to?

Reply:
Declaration in module file - https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java#L486
Usage - https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/ExchangeClientFactory.java#L58